src/common/interfaces.ts

```diff
@@ -19,15 +19,24 @@ export interface IHeartbeatConfig {
 
 export interface IJobAndTaskResponse {
   task: ITaskResponse<unknown>;
-  job: IJobResponse<PolygonPartsPayload, unknown>;
+  job: JobProfile;
 }
 
 export interface IJobHandler {
-  processJob: (job: IJobResponse<PolygonPartsPayload, unknown>) => Promise<void>;
+  processJob: (job: JobProfile) => Promise<PolygonPartsEntityName>;
 }
 
 export interface IPermittedJobTypes {
   ingestionNew: string;
   ingestionUpdate: string;
   ingestionSwapUpdate: string;
 }
+
+export interface JobParams {
+  metadata: Record<string, unknown>;
+  partsData: PolygonPart[];
+  inputFiles: InputFiles;
+  additionalParams: Record<string, unknown>;
+}
+
+export type JobProfile = IJobResponse<JobParams, unknown>;
```

Comment:
JobProfile is not that representing name , you can just call this type 'JobResponse'

Reply:
i would even use the `IJobResponse<JobParams, unknown>;` not sure we want a type here